src/NServiceBus.Core/Transports/OutgoingMessage.cs

```diff
@@ -0,0 +1,32 @@
+namespace NServiceBus.Transports
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// The message going out to the transport
+    /// </summary>
+    public class OutgoingMessage
+    {
+        /// <summary>
+        /// Constructs the message
+        /// </summary>
+        /// <param name="headers">The headers associated with this message</param>
+        /// <param name="body">The body of the message</param>
+        public OutgoingMessage(Dictionary<string, string> headers,byte[] body)
```

Comment:
Is it?

With the databus data is stored on "disk" and streamed into memory, all
good.

This is outgoing messages so it will already be in memory since the users
will have "new'ed" up the message? (so you will already have allocated all
the memory)

On Tue, Mar 10, 2015 at 8:44 AM, John Simons <notifications@github.com>
wrote:

> In src/NServiceBus.Core/Transports/OutgoingMessage.cs
> <https://github.com/Particular/NServiceBus/pull/2678#discussion_r26103254>
> :
>
> > @@ -0,0 +1,32 @@
> > +namespace NServiceBus.Transports
> > +{
> > +    using System.Collections.Generic;
> > +
> > +    /// <summary>
> > +    /// The message going out to the transport
> > +    /// </summary>
> > +    public class OutgoingMessage
> > +    {
> > +        /// <summary>
> > +        /// Constructs the message
> > +        /// </summary>
> > +        /// <param name="headers">The headers associated with this message</param>
> > +        /// <param name="body">The body of the message</param>
> > +        public OutgoingMessage(Dictionary<string, string> headers,byte[] body)
>
> I'm confused, what are u unclear about? This is exactly the same issue as
> the issue with the databus!
> … <#14c02a4c511f2712_>
> On Tuesday, March 10, 2015, Andreas Öhlund <notifications@github.com>
> wrote: In src/NServiceBus.Core/Transports/OutgoingMessage.cs <
> https://github.com/Particular/NServiceBus/pull/2678#discussion_r26103000>
> : > @@ -0,0 +1,32 @@ > +namespace NServiceBus.Transports > +{ > + using
> System.Collections.Generic; > + > + /// <summary> > + /// The message going
> out to the transport > + /// </summary> > + public class OutgoingMessage >
> + { > + /// <summary> > + /// Constructs the message > + /// </summary> > +
> /// <param name="headers">The headers associated with this message</param>
> > + /// <param name="body">The body of the message</param> > + public
> OutgoingMessage(Dictionary<string, string> headers,byte[] body)
> @johnsimons <https://github.com/johnsimons> <https://github.com/johnsimons>
> I could go both ways. How about we make this out of scope for this pull and
> you and me spike this when you're back next week? — Reply to this email
> directly or view it on GitHub <
> https://github.com/Particular/NServiceBus/pull/2678/files#r26103000>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/NServiceBus/pull/2678/files#r26103254>.
>